v2/internal/identity/credential_provider.go

```diff
@@ -300,3 +334,10 @@ func (c *credentialProvider) getSecret(ctx context.Context, namespace string, se
 func getSecretNameFromAnnotation(credentialFrom string, resourceNamespace string) types.NamespacedName {
 	return types.NamespacedName{Namespace: resourceNamespace, Name: credentialFrom}
 }
+
+func authModeOrDefault(mode string) AuthModeOption {
+	if mode == string(podIdentity) {
+		return podIdentity
+	}
+	return workloadIdentity
+}
```

Comment:
This treats anything other than `podIdentity` as `workloadIdentity` - including `foo`, `bang`, and `oh-oh`. 

A misconfigured value should at least trigger a log warning, if not an actual error. 

I'm also a fan of being case insensitive for configuration values, though @matthchr doesn't always agree.

```suggestion
func authModeOrDefault(mode string) (AuthModeOption, error) {
	if strings.EqualFold(mode, string(podIdentity)) {
		return podIdentity, nil
	}
	
	if strings.EqualFold(mode, string(workloadIdentity)) {
		return workloadIdentity, nil
	}
	
	return "", errors.Errorf("authorization mode %q not valid", mode)
}
```

Reply:
> I'm also a fan of being case insensitive for configuration values, though @matthchr doesn't always agree.

I don't have a philosophical objection here one way or the other. I do think that @theunrepentantgeek is correct we should have a warning or error if the user gets it wrong. I think from a consistency perspective, being case-sensitive here might make sense given other fields (such as enums in the CRDs themselves) are also case-sensitive, and case-sensitivity seems to be "the standard" in the Kubernetes world (look at labels which are case-sensitive, or names which only allow lowercase and thus are case-sensitive).

My dislike of case-insensitivity comes primarily from REST, where attempting to be case-insensitive is awkward because JSON keys are stereotypically case-sensitive (so the expectation is not necessarily case-insensitivity) and REST entities are both set by the user and returned to the user. In the "returned to the user" scenario, case insensitivity gets awkward because you want to preserve what they've sent you, but internally you want to store it in a single canonical form. Accomplishing this ends up being:
1. More work in your backend (need to remember their case)
2. Confusing. Which of their cases do you remember? The latest PUT? Or the first PUT? Or something else?

Postel's law ends up not _really_ applying because usually in REST you have to return exactly what you accept (the expectation is that what you PUT is what you GET, basically), so you can't actually be liberal in what you accept but conservative in what you return. They are one and the same. You have to pick a winner and I tend to favor picking the easier to implement (and simpler to explain) approach and just give an error in all other cases.